_hooks/pre-commit

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+#
+
+# A git hook script that checks for trailing whitespace
+# in your commits. Bypass it with the --no-verify option
+# to git-commit
+#
+# usage: make a soft link to this file,
+# e.g., ln -s ~/config/pre-commit.git.sh ~/some_project/.git/hooks/pre-commit
+
+if test "$(git diff --check --cached)"
```

Comment:
Ok, I understand what you are saying here now.  You'd like the pre-commit hook to not only check whitespace, but also run the other checks that are currently being run with travis and can be run locally, but with individual commands.  Let me see what I can stir up on this piece...